cmd/mdatagen/validate.go

```diff
@@ -109,6 +109,20 @@ func (md *metadata) validateResourceAttributes() error {
 		if attr.Type == empty {
 			errs = multierr.Append(errs, fmt.Errorf("empty type for resource attribute: %v", name))
 		}
+
+		for _, filter := range attr.Include {
+			if err := filter.Validate(); err != nil {
+				errs = multierr.Append(errs, fmt.Errorf(`resource attribute filter "%v": %w`, filter, err))
+				continue
+			}
+		}
+		for _, filter := range attr.Exclude {
+			if err := filter.Validate(); err != nil {
+				errs = multierr.Append(errs, fmt.Errorf(`resource attribute filter "%v": %w`, filter, err))
+				continue
+			}
+		}
```

Comment:
This is probably not needed. https://github.com/open-telemetry/opentelemetry-collector/blob/e7e73ab2755d9628089aa0cdaba8e2e68e31d604/component/config.go#L90 should take care of this. But please verify